### ISSUE TITLE
FCS witnesses issue

### ISSUE THREAD

**Issue by ncave** [dotnet/fsharp#10408]
Typechecking the following in FCS with `--langversion:preview` returns an error:
```fs
let test x =
    match x with
    | [| |] -> x
    | _ -> x
```
The error is thrown in [GetWitnessArgs](https://github.com/dotnet/fsharp/blob/main/src/fsharp/symbols/Exprs.fs#L499), resulting in the following exception:
```
[ERROR] The exception has been reported. This internal exception should now be caught at an error recovery point on the stack. Original message: Exception of type 'FSharp.Compiler.ConstraintSolver+ConstraintSolverMissingConstraint' was thrown.)
   at FSharp.Compiler.ErrorLogger.ErrorLoggerExtensions.ErrorLogger.Error[b](ErrorLogger x, Exception exn) in /fsharp/src/fsharp/ErrorLogger.fs:line 414
   at FSharp.Compiler.ErrorLogger.CommitOperationResult[a](OperationResult`1 res) in /fsharp/src/fsharp/ErrorLogger.fs:line 552
   at FSharp.Compiler.SourceCodeServices.FSharpExprConvert.GetWitnessArgs(SymbolEnv cenv, ExprTranslationEnv env, ValRef vref, range m, FSharpList`1 tps, FSharpList`1 tyargs) in /fsharp/src/fsharp/symbols/Exprs.fs:line 499
```

**Comment by cartermp** [comment]
@dsyme FYI

**Comment by cartermp** [comment]
Note that this is strictly for FCS consumers like Fable; "normal" F# code does not exhibit this issue

**Comment by dsyme** [comment]
@ncave thanks for the report.  Strange that the code doesn't involve either witness or constraints.  Anyway, we'll fix it.

**Comment by alfonsogarciacaro** [comment]
FYI, after building FCS with PR #11388 merged and removing the [code we had to ignore Witnesses errors](https://github.com/ncave/fsharp/pull/2/commits/0c690987a0cfaed1db6a2a3f9961c938e8c65263) Fable tests are passing which is great news! However, when trying to self-compile Fable we still have an error when compiling TypedTree.fs, which means something is still wrong and we had to add back the code to ignore witnesses errors. See https://github.com/fable-compiler/Fable/pull/2444#issuecomment-842780351

I'll try to investigate more.
